client/trino-client/src/main/java/io/trino/client/StatementClientV1.java

```diff
@@ -111,10 +115,10 @@ public StatementClientV1(Call.Factory httpCallFactory, Call.Factory segmentHttpC
     {
         requireNonNull(httpCallFactory, "httpCallFactory is null");
         requireNonNull(session, "session is null");
-        requireNonNull(query, "query is null");
```

Comment:
We can't skip this check.  If we go this route, this would have to be a private constructor and there should be 2 public ones: one that takes a query and makes sure it's not null and another that does not take a query, so that the intent of the caller is clear.

But after thinking more about it, I feel it would be best to avoid null queries altogether.  Instead of changing this class to make it usable for validation, we could refactor the `createRequestBuilder()` functionality into a stand-alone class and use that for the validation requests, since that is really the only functionality needed for connection validation.

Reply:
I saw two options here.

The first one is to keep clients as stubs of their server side functions. In this case, executing query and validating credential shall stay in the same class as it is now. To reduce the confusion of `query`, we can remove `query` from the construction and add a function `startQuery`. I prefer this code style, constructing an object then explicitly putting it in motion, instead of creating an object in motion.

The other options is to define client objects based on client functionalities. In this case, we should create `AuthenticationClient` as @aalbu suggested.

I think Trino maintainers do have plans to add more endpoints in the future so I am leaning to the first option and keep the 2nd options as Trino adding more endpoints.